supybot_fedora/plugin.py

```diff
@@ -195,46 +213,104 @@ def __init__(self, irc):
         # fedmsg.meta.make_processors(**fm_config)
 
     def _refresh(self):
-        timeout = socket.getdefaulttimeout()
-        socket.setdefaulttimeout(None)
         self.log.info("Downloading user data")
-        try:
-            request = self.fasclient.send_request(
-                "/user/list", req_params={"search": "*"}, auth=True, timeout=240
-            )
-            users = request["people"] + request["unapproved_people"]
-            del request
-        except AuthError:
-            self.log.info("Error Authorizing to FAS")
-            users = []
-
-        self.log.info("Caching necessary user data")
-        self.users = {}
-        self.faslist = {}
-        self.nickmap = {}
-        for user in users:
-            name = user["username"]
-            self.users[name] = {}
-            self.users[name]["id"] = user["id"]
-            key = " ".join(
-                [
+
+        if self.registryValue("use_fasjson"):
+            self.log.info("Caching necessary user data")
+            self.users = []
+            self.faslist = {}
+            self.nickmap = {}
+            users = list(self.fasjsonclient.list_all_entities("users"))
+            for user in users:
+                name = user["username"]
+                self.users.append(name)
+                key = " ".join(
+                    [
+                        user["username"],
+                        user["emails"][0],
+                        user["givenname"] or "",
+                        user["surname"] or "",
+                        user["ircnicks"][0] or "",
+                    ]
+                )
+                value = "%s '%s %s' <%s>" % (
+                    user["username"],
+                    user["givenname"] or "",
+                    user["surname"] or "",
+                    user["emails"][0] or "",
+                )
+                self.faslist[key] = value
+                if user["ircnicks"]:
+                    for nick in user["ircnicks"]:
```

Comment:
You can do `for nick in user.get("ircnicks", []):` and you'll avoid the `if` condition. Less indentation! :-)

Reply:
fixed